core/src/main/java/com/linecorp/armeria/internal/client/AbstractRuleBuilder.java

```diff
@@ -52,16 +51,19 @@
     @Nullable
     private Predicate<Throwable> exceptionFilter;
 
+    /**
+     * Creates a new instance with the specified {@code requestHeadersFilter}.
+     */
     @SuppressWarnings("unchecked")
-    AbstractRetryRuleBuilder(Predicate<? super RequestHeaders> requestHeadersFilter) {
+    protected AbstractRuleBuilder(Predicate<? super RequestHeaders> requestHeadersFilter) {
         this.requestHeadersFilter = (Predicate<RequestHeaders>) requestHeadersFilter;
     }
 
     /**
      * Adds the specified {@code responseHeadersFilter} for a {@link RetryRule} which will retry
```

Comment:
This class is not just for `RetryRule` anymore.

Reply:
Yeap, I cleaned up Javadoc.